build/proxies-scenarios.yml

```diff
@@ -96,17 +109,30 @@ steps:
+      - ${{ if and( or( not( protocol.requiresServerHttps), s.supportsServerHttps), or( not( protocol.requiresServerHttp2), s.supportsServerHttp2) ), or( not( protocol.requiresDownstreamHttp2), s.supportsDownstreamHttp2) ), or( not( protocol.requiresServerGRPC), s.supportsGRPC) ) }}:
+        - ${{ if not(s.requiresGRPC) }}:
+          # For .NET 5.0, only run HTTP-HTTP and H2-H2 benchmarks
+          - ${{ if or( not( eq( s.displayName, 'YARP-net50')), in( protocol.displayName, 'http - http', 'h2 - h2') ) }}:
+            - task: PublishToAzureServiceBus@1
+              condition: succeededOrFailed()
+              displayName: ${{ s.displayName }} ${{ payload.displayName }} ${{ protocol.displayName }}
+              inputs:
+                connectedServiceName: ${{ parameters.connection }}
+                waitForCompletion: true
+                messageBody: |
+                  {
+                    "name": "crank",
+                    "args": [ "--table ProxyBenchmarks --sql SQL_CONNECTION_STRING --chart --session $(session) --profile ${{ parameters.profile }} --no-metadata --no-measurements --downstream.options.reuseBuild true --load.variables.warmup ${{ parameters.warmup }} --load.variables.duration ${{ parameters.duration }} ${{ s.arguments }} ${{ payload.arguments }} ${{ protocol.arguments }} --description \"${{ s.displayName }} ${{ payload.displayName }} ${{ protocol.displayName }} ${{ parameters.profile }}\"" ]
+                  }
+        - ${{ if s.requiresGRPC }}:
           - task: PublishToAzureServiceBus@1
             condition: succeededOrFailed()
-            displayName: ${{ s.displayName }} ${{ payload.displayName }} ${{ protocol.displayName }}
+            displayName: ${{ s.displayName }} ${{ protocol.displayName }}
             inputs:
               connectedServiceName: ${{ parameters.connection }}
               waitForCompletion: true
               messageBody: |
                 {
                   "name": "crank",
-                  "args": [ "--table ProxyBenchmarks --sql SQL_CONNECTION_STRING --chart --session $(session) --profile ${{ parameters.profile }} --no-metadata --no-measurements --downstream.options.reuseBuild true --load.variables.warmup ${{ parameters.warmup }} --load.variables.duration ${{ parameters.duration }} ${{ s.arguments }} ${{ payload.arguments }} ${{ protocol.arguments }} --description \"${{ s.displayName }} ${{ payload.displayName }} ${{ protocol.displayName }} ${{ parameters.profile }}\"" ]
+                  "args": [ "--table ProxyBenchmarks --sql SQL_CONNECTION_STRING --chart --session $(session) --profile ${{ parameters.profile }}-grpc --no-metadata --no-measurements --downstream.options.reuseBuild true --load.variables.warmup ${{ parameters.warmup }} --load.variables.duration ${{ parameters.duration }} ${{ s.arguments }} ${{ protocol.arguments }} --description \"${{ s.displayName }} ${{ protocol.displayName }} ${{ parameters.profile }}\"" ]
```

Comment:
why the `-grpc` after `--profile ${{ parameters.profile }}`

Reply:
Ah, I realized that it's wrong. I need to add `-grpc` to scenario name, not profile name. The goal is to show grpc as yet another protocol like h2. Thus, I tried to keep the existing scenarios (like `YARP`), but then patch `--scenario proxy-yarp` argument with an extra `-grpc` suffix to call the right scenario.

Reply:
Did I get it right that the proxy to be shown on the dashboards is identified by `--property proxy=...` and not by `displayName`? I've added new scenarios like "YARP gRPC" with different `--scenario` values, but the same ``--property proxy` value. That also eliminates the second `if` block.